kernel/src/snapshot.rs

```diff
@@ -321,47 +321,63 @@ fn read_last_checkpoint(
 
 /// List all log files after a given checkpoint.
 fn list_log_files_with_checkpoint(
-    cp: &CheckpointMetadata,
+    checkpoint_metadata: &CheckpointMetadata,
     fs_client: &dyn FileSystemClient,
     log_root: &Url,
 ) -> DeltaResult<(Vec<FileMeta>, Vec<FileMeta>)> {
-    let version_prefix = format!("{:020}", cp.version);
+    let version_prefix = format!("{:020}", checkpoint_metadata.version);
     let start_from = log_root.join(&version_prefix)?;
 
-    let files = fs_client
-        .list_from(&start_from)?
-        .collect::<Result<Vec<_>, Error>>()?
-        .into_iter()
+    let mut max_checkpoint_version = checkpoint_metadata.version;
+    let mut checkpoint_files = Vec::with_capacity(10);
```

Comment:
Just curious, any reason why you expect the `Vec` to be at most size 10?

Reply:
That is odd, because most checkpoints will have just one file?
I wonder if the intent was to pre-size the `commit_files` to 10, since that's the default checkpoint interval for delta-spark?

Reply:
Yeah, that makes sense. I've swapped them so we reserve 10 for commits.